src/military/military.cpp

```diff
@@ -1021,12 +1021,18 @@ void regenerate_land_unit_average(sys::state& state) {
 	*/
 	auto const max = state.military_definitions.unit_base_definitions.size();
 	state.world.for_each_nation([&](dcon::nation_id n) {
-		float total = 0;
-		float count = 0;
-
 		auto lo_mod = state.world.nation_get_modifier_values(n, sys::national_mod_offsets::land_attack_modifier);
 		auto ld_mod = state.world.nation_get_modifier_values(n, sys::national_mod_offsets::land_defense_modifier);
 
+		auto irregular = state.military_definitions.irregular;
+		auto& irreg_stats = state.world.nation_get_unit_stats(n, irregular);
+
+		// handle irregular infantry as it seems to be unhandled by code below...
```

Comment:
why do you believe that the loop below doesn't cover irregulars? An irregular is one of the unit types, and so it should appear as part of the iteration

Reply:
For some reason, all unciv nations had 0 military score with undefined average stats before this change. Was this behavior unique to my copy?

Reply:
Ah,  yes, there is a bug in the body of the loop. It should check if the unit has been activated *or if it is active from the start of the game* (irregulars being one of the units covered by the second part of that check)